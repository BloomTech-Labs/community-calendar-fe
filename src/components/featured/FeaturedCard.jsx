import React from 'react'
import moment from 'moment'
import {Link} from 'react-router-dom'
import PropTypes from 'prop-types'

//styles
import {
  featuredCard,
  featuredCardContent,
  eventImage,
} from './Featured.module.scss'

/*
Cards for the Featured Events carousel.
*/

export default function FeaturedCard(props) {
  const {item} = props
  let location = item.locations.length
    ? item.locations[item.locations.length - 1]
    : null

  return (
    <Link className={`${featuredCard}`} to={`events/${item.id}`}>
      {item.eventImages.length > 0 && (
        <img
          className={eventImage}
          src={item.eventImages[0].url}
          alt='Event image'
        />
      )}

      <div className={featuredCardContent}>
        <p>
          <time
            className='subtitle is-7'
            dateTime={moment(item.start).format('YYYY-MM-DD')}
          >
            {moment(item.start).format('MMMM DD, YYYY')}
          </time>
        </p>
        <p
          className='is-6 Butler has-text-weight-bold'
          style={{height: '48px'}}
        >
          {item.title}
        </p>
        <p>
          <time
            className='subtitle is-size-7'
            dateTime={moment(item.start).format('HH:mm')}
          >
            {moment(item.start).format('h:mm a')}
          </time>
          &nbsp;&#8208;&nbsp;
          <time
            className='subtitle is-size-7'
            dateTime={moment(item.end).format('HH:mm')}
          >
            {moment(item.end).format('h:mm a')}
          </time>
        </p>
        {location && (
          <p className='is-size-7 has-text-weight-bold'>
            <span className='has-text-weight-light'>Location: </span>
            {`${location.streetAddress}, ${location.city}`}
          </p>
        )}
      </div>
    </Link>
  )
}

FeaturedCard.propTypes = {
  item: PropTypes.object,
}
