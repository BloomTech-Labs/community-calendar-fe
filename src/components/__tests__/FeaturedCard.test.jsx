import React from 'react'
// Testing libraries
import {render} from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import userEvent from '@testing-library/user-event'

// Modules used in components
import {MemoryRouter} from 'react-router-dom'
import moment from 'moment'

// Components to test
import FeaturedCard from '../featured/FeaturedCard'

//mock TEST_FULL
import {TEST_FULL} from '../../../test-utils/mock_data/test_event'

describe('Tests for FeaturedCards', () => {
  beforeEach(() => {
    // reset mock functions before each test
    jest.clearAllMocks()
  })

  test('should match snapshot', () => {
    jest.mock('moment', () => () => ({format: () => 'test'}))

    const tree = render(
      <MemoryRouter>
        <FeaturedCard item={TEST_FULL} />
      </MemoryRouter>,
    )
    expect(tree).toMatchSnapshot()
  })

  test('Should display correct TEST_FULL', () => {
    jest.mock('moment', () => () => ({format: () => 'test'}))
    const {getByText} = render(
      <MemoryRouter>
        <FeaturedCard item={TEST_FULL} />
      </MemoryRouter>,
    )

    expect(getByText(TEST_FULL.title)).toBeInTheDocument()
    expect(getByText(/600 woodward avenue, detroit/i)).toBeInTheDocument()
  })
})
