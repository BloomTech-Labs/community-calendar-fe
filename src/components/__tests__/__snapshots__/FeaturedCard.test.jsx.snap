// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for FeaturedCards should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        class="featuredCard"
        href="/events/ck5flbejv00200730x34y4le6"
      >
        <img
          alt="Event image"
          class="eventImage"
          src="https://img.evbuc.com/https%3A%2F%2Fcdn.evbuc.com%2Fimages%2F69917729%2F89373299245%2F1%2Foriginal.jpg?w=800&auto=format%2Ccompress&q=75&sharp=10&rect=0%2C693%2C4000%2C2000&s=94184073d91c25306d22d1b0401cfea9"
        />
        <div
          class="featuredCardContent"
        >
          <p>
            <time
              class="subtitle is-7"
              datetime="2020-01-22"
            >
              January 22, 2020
            </time>
          </p>
          <p
            class="is-6 Butler has-text-weight-bold"
            style="height: 48px;"
          >
            Art & Architecture - Downtown Walking Tour
          </p>
          <p>
            <time
              class="subtitle is-size-7"
              datetime="09:00"
            >
              9:00 am
            </time>
             ‐ 
            <time
              class="subtitle is-size-7"
              datetime="11:30"
            >
              11:30 am
            </time>
          </p>
          <p
            class="is-size-7 has-text-weight-bold"
          >
            <span
              class="has-text-weight-light"
            >
              Location: 
            </span>
            600 Woodward Avenue, Detroit
          </p>
        </div>
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      class="featuredCard"
      href="/events/ck5flbejv00200730x34y4le6"
    >
      <img
        alt="Event image"
        class="eventImage"
        src="https://img.evbuc.com/https%3A%2F%2Fcdn.evbuc.com%2Fimages%2F69917729%2F89373299245%2F1%2Foriginal.jpg?w=800&auto=format%2Ccompress&q=75&sharp=10&rect=0%2C693%2C4000%2C2000&s=94184073d91c25306d22d1b0401cfea9"
      />
      <div
        class="featuredCardContent"
      >
        <p>
          <time
            class="subtitle is-7"
            datetime="2020-01-22"
          >
            January 22, 2020
          </time>
        </p>
        <p
          class="is-6 Butler has-text-weight-bold"
          style="height: 48px;"
        >
          Art & Architecture - Downtown Walking Tour
        </p>
        <p>
          <time
            class="subtitle is-size-7"
            datetime="09:00"
          >
            9:00 am
          </time>
           ‐ 
          <time
            class="subtitle is-size-7"
            datetime="11:30"
          >
            11:30 am
          </time>
        </p>
        <p
          class="is-size-7 has-text-weight-bold"
        >
          <span
            class="has-text-weight-light"
          >
            Location: 
          </span>
          600 Woodward Avenue, Detroit
        </p>
      </div>
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
